exec/java-exec/src/main/java/org/apache/drill/exec/work/WorkManager.java

```diff
@@ -277,7 +277,9 @@ public void startFragmentPendingRemote(final FragmentManager fragmentManager) {
         @Override
         protected void cleanup() {
           runningFragments.remove(fragmentHandle);
-          workBus.removeFragmentManager(fragmentHandle);
+          if (!fragmentManager.isCancelled()) {
+            workBus.removeFragmentManager(fragmentHandle, false);
```

Comment:
Not sure why you don't want to cancel here.

Reply:
If cleanup is not called as a result of FragmentManager cancellation, it is part of the regular cleanup after run is complete.